dace/transformation/interstate/if_raising.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+""" If raising transformation """
+
+from dace import data as dt, sdfg as sd
+from dace.sdfg import InterstateEdge
+from dace.sdfg import utils as sdutil
+from dace.sdfg.state import SDFGState
+from dace.transformation import transformation
+from dace.properties import make_properties
+
+@make_properties
+class IfRaising(transformation.MultiStateTransformation):
+    """
+    Duplicates an if guard and anticipates the evaluation of the condition
+    """
+
+    if_guard = transformation.PatternNode(sd.SDFGState)
+
+    @staticmethod
+    def annotates_memlets():
+        return False
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.if_guard)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
+        if_guard: SDFGState = self.if_guard
+
+        out_edges = graph.out_edges(if_guard)
+
+        if len(out_edges) != 2:
+            return False
+
+        if if_guard.is_empty():
+            return False
+
+        # check that condition does not depend on computations in the state
```

Comment:
might be missing something I do not know about here. Also I don't know if this can result in spurious failures